Framework/PythonInterface/plugins/algorithms/AlignComponents.py

```diff
@@ -189,47 +209,48 @@ def PyInit(self):
-        self.setPropertyGroup("Yposition", "Translation")
-        self.setPropertyGroup("MinYposition", "Translation")
-        self.setPropertyGroup("MaxYposition", "Translation")
-        self.setPropertyGroup("Zposition", "Translation")
-        self.setPropertyGroup("MinZposition", "Translation")
-        self.setPropertyGroup("MaxZposition", "Translation")
-
-        # Rotation
-        self.setPropertyGroup("EulerConvention", "Rotation")
-        self.setPropertyGroup("AlphaRotation", "Rotation")
-        self.setPropertyGroup("MinAlphaRotation", "Rotation")
-        self.setPropertyGroup("MaxAlphaRotation", "Rotation")
-        self.setPropertyGroup("BetaRotation", "Rotation")
-        self.setPropertyGroup("MinBetaRotation", "Rotation")
-        self.setPropertyGroup("MaxBetaRotation", "Rotation")
-        self.setPropertyGroup("GammaRotation", "Rotation")
-        self.setPropertyGroup("MinGammaRotation", "Rotation")
-        self.setPropertyGroup("MaxGammaRotation", "Rotation")
+        properties = ["EulerConvention", "AlphaRotation", "MinAlphaRotation", "MaxAlphaRotation",
+                      "BetaRotation", "MinBetaRotation", "MaxBetaRotation",
+                      "GammaRotation", "MinGammaRotation", "MaxGammaRotation"]
+        [self.setPropertyGroup(name, "Rotation") for name in properties]
 
     def validateInputs(self):
         """
         Does basic validation for inputs
         """
         issues = dict()
 
-        calWS: TableWorkspace = self.getProperty('CalibrationTable').value
+        peak_positions = self.getProperty('PeakPositions').value
+
+        table_tof: TableWorkspace = self.getProperty('PeakCentersTofTable').value
+
+        if 'detid' not in table_tof.getColumnNames():
+            issues['PeakCentersTofTable'] = 'PeakCentersTofTable is missing column "detid"'
+
+        # The titles for table columns storing the TOF peak-center positions start with '@'
+        column_names = [name for name in table_tof.getColumnNames() if name[0] == '@']
+        peak_count = len(peak_positions)  # number of reference peak-center values
+        if len(column_names) != peak_count:
+            error_message = f'The number of table columns containing the peak center positions' \
+                            f' {len(column_names)} is different than the number of peak positions {peak_count}'
+            issues['PeakCentersTofTable'] = error_message
 
-        if 'difc' not in calWS.getColumnNames() or 'detid' not in calWS.getColumnNames():
-            issues['CalibrationTable'] = "Calibration table requires detid and difc"
+        # The titles for table columns storing the TOF peak-center positions do contain the values
+        # of the reference peak-center positions in d-spacing units, up to a precision of 5
+        def with_precision(the_number, precision):
+            r"""Analog of C++'s std::setprecision"""
+            return round(the_number, precision - len(str(int(the_number))))
```

Comment:
Can we add a new line after the nested function?  Just to make it a little bit to read.